packages/flutter/test/material/carousel_test.dart

```diff
@@ -1977,6 +1978,299 @@ void main() {
     });
   });
 
+  group('CarouselView onIndexChanged callback', () {
+    testWidgets('CarouselView shows correct item after animation', (WidgetTester tester) async {
+      final controller = CarouselController();
+      addTearDown(controller.dispose);
+      var leadingIndex = 0;
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: CarouselView(
+              itemExtent: 300,
+              controller: controller,
+              itemSnapping: true,
```

Comment:
```suggestion
```

Removing snap feature to keep test simple and straightforward for the newly-added feature.